RoundsWithFriends/Algorithms/PickOrder.cs

```diff
@@ -0,0 +1,129 @@
+                this.cycleRunningIndex++;
+                return player;
+            }
+        }
+
+        private List<TeamPriority> priorities; // list of team priorities, order is NOT linked to teamID
+        private Dictionary<int, List<Player>> teams; // players in teams keyed by teamID
+        public PickOrder(List<Player> players)
+        {
+            this.teams = new Dictionary<int, List<Player>>() { };
+
+            this.priorities = new List<TeamPriority>();
+            foreach (Player player in players) 
+            {
+                if (!this.teams.ContainsKey(player.teamID)) { this.teams[player.teamID] = new List<Player>() { }; }
+
+                this.teams[player.teamID].Add(player);
+            }
+
+            foreach (int teamID in this.teams.Keys)
+            {
+                TeamPriority team = new TeamPriority(teamID);
+                this.priorities.Add(team);
+
+                foreach (Player player in this.teams[teamID])
+                {
+                    team.AddPlayer(player);
+                }
+            }
+
+            this.Reprioritize();
+        }
+
+        private void Reprioritize()
+        {
+            // Sort teams based on the most prioritized player of each team
+            this.priorities.Sort((a, b) => b.priority - a.priority);
+        }
+
+        public List<Player> GetPickOrder(int winningTeam)
+        {
+            var list = new List<Player>(); // List of players
+            int maxTeamPlayers = this.teams.Select(kv => kv.Value.Count()).Max();
+
+            for (int teamIndex = 0; teamIndex < this.priorities.Count; teamIndex++)
+            {
+                if (this.priorities[teamIndex].teamId != winningTeam)
+                {
+                    this.priorities[teamIndex].priority += teamIndex * teamIndex;
+                }
+            }
+
+            for (int playerIndex = 0; playerIndex < maxTeamPlayers; playerIndex++)
+            {
+                for (int teamIndex = 0; teamIndex < this.priorities.Count; teamIndex++)
+                {
+                    if (playerIndex >= this.priorities[teamIndex].players.Count || this.priorities[teamIndex].teamId == winningTeam)
+                    {
+                        continue;
+                    }
+
+                    list.Add(this.priorities[teamIndex].GetNextPlayer());
+                }
+            }
+
+            int firstTeamIndex = winningTeam == this.priorities[0].teamId ? 1 : 0;
+
+            for (int teamIndex = 0; teamIndex < this.priorities.Count; teamIndex++)
+            {
+                if (teamIndex == firstTeamIndex)
+                {
+                    this.priorities[teamIndex].NextCycle();
+                }
+                else if (this.priorities[teamIndex].teamId != winningTeam)
+                {
+                    this.priorities[teamIndex].ResetCycle();
+                }
+            }
+
+            this.Reprioritize();
+            return list;
+        }
```

Comment:
If you take a subset of `priorities` here, omitting the winning team, you can remove a lot of the conditional logic here.
```csharp
var filteredPriorities = this.priorities.Where(p => p.teamId != winningTeam).ToList();
```
Now you don't need to check for it, and the first team to pick is always `filteredPriorities[0]`, for example.